### PR TITLE
Update assembly versions when servicing

### PR DESCRIPTION
**Pull request by dougbu**
- #24540
- remove `$(DisableServicingFeatures)` property; `$(IsServicingBuild)` should work now

### CODE REVIEW

**Comment by dougbu** [comment]
@Pilchie what's the correct way to let Steve and maybe all of Tactics know about real changes like this❔ Apparently we have a _bit_ of room for infrastructure and test-only changes in RC2 before the end of the week.

@dotnet/aspnet-build this change has been validated in my 5.0.1 fake build. There are two additional real changes and at least two test-only changes that'll come out of the servicing exercise. 5.0.0 is now testing cleanly but I have more validation to do before I can be confident that's everything.

**Review by JunTaoLuo — APPROVED** [review]
Seems reasonable. Let me know if verification leads to additional changes though.

File: Directory.Build.targets [hunk 1]
```diff
@@ -130,8 +130,14 @@
...
-  <PropertyGroup Condition="'$(Language)' == 'C#'">
-    <AssemblyVersion>$(AspNetCoreMajorVersion).$(AspNetCoreMinorVersion).0.0</AssemblyVersion>
+  <!--
+    If assembly is in shared framework and not net4*, assembly version should remain fixed for an entire
+    major.minor release even in servicing.
+  -->
+  <PropertyGroup Condition=" '$(IsAspNetCoreApp)' == 'true' AND
+      '$(Language)' == 'C#' AND
+      '$(TargetFrameworkIdentifier)' != '.NETFramework' ">
+    <AssemblyVersion>$(AspNetCoreMajorMinorVersion).0.0</AssemblyVersion>
```
Comment (wtgodbe) [inline Directory.Build.targets:140]: What controls netfx assembly versions now?

**Review by wtgodbe — APPROVED** [review]

File: Directory.Build.targets (hunk 1 above)
Comment (dougbu) [reply Directory.Build.targets:140]: Usual Arcade auto-updating version is used for .NET framework assemblies and non-`$(IsAspNetCoreApp)` projects

**Pull request merged**